src/adapters/nedb-search-index/index.ts

```diff
@@ -1,32 +1,76 @@
 import { Logger } from '@navarik/types'
 import Database from '@navarik/nedb'
-import { SearchIndex, SearchQuery, SearchOptions, CanonicalEntity, ActionType } from '../../types'
+import { SearchIndex, SearchQuery, SearchOptions, CanonicalEntity, ActionType, CanonicalSchema } from '../../types'
 import { NeDbQueryParser } from './ne-db-query-parser'
+import { FullTextFieldExtractor } from './fulltext-field-extractor'
 
-
-type Config = {
+interface Config {
   logger: Logger
 }
 
-const databaseError = (err: Error) => {
-  throw new Error(`[NeDB] Database error: ${err.message}`)
+const callback = (resolve, reject) => (err: Error, res) => {
+  if (err) {
+    reject(new Error(`[NeDB] Database error: ${err.message}`))
+  } else {
+    resolve(res)
+  }
 }
 
 export class NeDbSearchIndex<M extends object> implements SearchIndex<M> {
   private logger: Logger
-  private client: Database
+  private documents: Database
+  private fullText: Database
   private queryParser: NeDbQueryParser
+  private fullTextFieldExtractor: FullTextFieldExtractor
 
   constructor({ logger }: Config) {
     this.logger = logger
-    this.client = new Database()
-    this.queryParser = new NeDbQueryParser()
-    this.client.ensureIndex({ fieldName: 'id', unique: true })
+    this.documents = new Database()
+    this.documents.ensureIndex({ fieldName: 'id', unique: true })
+    this.fullText = new Database()
+    this.queryParser = new NeDbQueryParser({ documentsDb: this.documents, fullTextDb: this.fullText })
+    this.fullTextFieldExtractor = new FullTextFieldExtractor({ callback: this.onFullTextUpdate.bind(this) })
```

Comment:
Since this thing is pretty memory intensive, would it make sense to pass in the extractor somehow? I'd like to be able to no-op it if I knew my application wasn't going to perform full text search.

Reply:
Discussed this, and it's a "maybe later, if we need it" kind of thing.